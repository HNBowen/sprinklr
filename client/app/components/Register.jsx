import React from 'react'
import PropTypes from 'prop-types'

const Register = (props) => {
  return (
      <div>
        <form onSubmit={(e) => props.handleRegister(e)}>
          username: <input type="text" name="username"></input>
          password: <input text="text" name="password"></input>
          <button type="submit">Register</button>
        </form>
        <a href="/login">Already have an account? Click here to login.</a>
      </div>
    )
}

Register.propTypes = {
  handleRegister: PropTypes.func.isRequired
}

export default Register