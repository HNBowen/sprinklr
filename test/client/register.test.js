import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import Register from '../../client/app/components/Register.jsx'



describe('Register', function() {

  let wrapper;
  beforeEach(function() {
    wrapper = shallow(<Register handleRegister={jest.fn()}/>);
  });

  it('should render correctly', function() {
    jestExpect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render with a form field', function() {
    expect(wrapper.find('form').length).to.equal(1)
  })

  it('should render with input fields for username and password', function() {
    expect(wrapper.find('input').length).to.equal(2)
    expect(wrapper.find('input').get(0).props.name).to.equal('username')
    expect(wrapper.find('input').get(1).props.name).to.equal('password')
  })

  it('should render with a submit button that reads "Register"', function() {
    expect(wrapper.find('button').length).to.equal(1)
    expect(wrapper.find('button').at(0).text()).to.equal('Register')
    expect(wrapper.find('button').get(0).props.type).to.equal('submit')
  })

  it('should call handleRegister when form is submitted', function() {
    let mockHandler = jest.fn();
    wrapper = shallow(<Register handleRegister={mockHandler} />)

    wrapper.find('form').simulate('submit')
    expect(mockHandler.mock.calls.length).to.equal(1);
  })

  it('should rende with an anchor tag linking to login page', function() {
    expect(wrapper.find('a[href="/login"]')).to.exist
    expect(wrapper.find('a[href="/login"]').text()).to.equal('Already have an account? Click here to login.')
  })
})