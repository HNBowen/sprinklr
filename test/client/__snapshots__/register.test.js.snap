// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    username: 
    <input
      name="username"
      type="text"
    />
    password: 
    <input
      name="password"
      text="text"
    />
    <button
      type="submit"
    >
      Register
    </button>
  </form>
  <a
    href="/login"
  >
    Already have an account? Click here to login.
  </a>
</div>
`;
